### PR TITLE
Initial Debezium Outbox support

### PR DESCRIPTION
**Pull request by mjholder**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-34508

## Description of Intent of Change(s)
The what, why and how.
This adds a new outbox model for debezium to read. It also adds a new debezium connector template to pickup changes in the outbox.
## Local Testing
How can the feature be exercised?
How can the bug be exploited and fix confirmed?
Is any special local setup required?

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: rbac/management/migrations/0047_outbox.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                        default=uuid.uuid4,
+                        editable=False,
+                        primary_key=True,
+                        serialize=False,
+                    ),
+                ),
+                ("aggregatetype", models.CharField(max_length=255)),
+                ("aggregateid", models.CharField(max_length=255)),
+                ("event_type", models.CharField(db_column="type", max_length=255)),
+                ("payload", models.BinaryField()),
```
Comment (lpichler) [inline rbac/management/migrations/0047_outbox.py:32]: I am sorry for confusion, we need jsonb here in the end - https://debezium.io/documentation/reference/stable/transformations/outbox-event-router.html#basic-outbox-table

discussed [here](https://redhat-internal.slack.com/archives/C059NUD49PD/p1723550716218259)

**Comment by lpichler** [comment]
@mjholder maybe can you do rebase ? not sure why CI is failing.

**Comment by lpichler** [comment]
@mjholder if CI failure is related to connector, can outbox table/model be done in separate PR? 🙏

**Comment by mjholder** [comment]
> @mjholder if CI failure is related to connector, can outbox table/model be done in separate PR? 🙏

Weird. I'll put it in a separate pr

**Review by lpichler — APPROVED** [review]

**Comment by lpichler** [comment]
NOTE: Table is defined in https://debezium.io/documentation/reference/stable/transformations/outbox-event-router.html#basic-outbox-table

**Pull request merged**